### PR TITLE
Auto-Connect port stuff

### PR DESCRIPTION
**Pull request by DonLakeFlyer**
- Centralize board detection in QGCSerialPortInfo class. This also hides the android/non-android differences from consumers.
- Change code to use new QGCSerialPortInfo board detection logic.
- Add new requiresUSBMavlinkStart method to LinkInterface. This signals whether the mavlink start shell command is needed.
- Update MAVLinkProtocol to use requiresUSBMavlinkStart appropriately.

### CODE REVIEW

**Comment by DonLakeFlyer** [comment]
@dogmaphobic Can you check the QGCSerialPortInfo and LinkManager changes? This is in the middle of everything and needs more eyes. I tested by flashing and connecting to all the types of stuff I own.

**Comment by DonLakeFlyer** [comment]
Another step into the maelstrom of Auto-Connect...

**Comment by DonLakeFlyer** [comment]
Also FYI: The current state of this code sets preferred on all auto-detected board types. So they will all show up at the top. Usually not a problem since you normally only have on connection. If you happen to have more than one thing connected, they will all go to the top above your persisted connections. I think thats probably ok.

**Comment by DonLakeFlyer** [comment]
I'm going to merge this so I can make forward progress. @dogmaphobic Can you still double check it after the fact?

File: src/comm/LinkManager.cc [hunk 1]
```diff
@@ -504,8 +499,15 @@ void LinkManager::_updateConfigurationList(void)
 #endif
         // Save port name
         currentPorts << portInfo.systemLocation();
-        // Is this a PX4 and NOT in bootloader mode?
-        if (portInfo.vendorIdentifier() == SerialPortIds::px4VendorId && !portInfo.description().contains("BL")) {
+
+        QGCSerialPortInfo::BoardType_t boardType = portInfo.boardType();
+
+        if (boardType != QGCSerialPortInfo::BoardTypeUnknown) {
+            if (portInfo.isBootloader()) {
```
Comment (thomasgubler) [inline src/comm/LinkManager.cc]: I noticed this when verifying #2221 but I am not sure if it's intended: isBootloader() calls boardTypePixhawk() which calls boardType() hence the whole board type logic is done twice in quick succession (see line 503).

Comment (DonLakeFlyer) [inline src/comm/LinkManager.cc]: I could probably make boardType return boot loader status to remove the double call. Let me see.

**Comment by DonLakeFlyer** [comment]
Thanks for verifying @thomasgubler

**Pull request merged**